clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -389,6 +397,11 @@ else if (metadata.error() == Errors.TOPIC_AUTHORIZATION_FAILED)
                 log.debug("Updating last seen epoch for partition {} from {} to epoch {} from new metadata", tp, currentEpoch, newEpoch);
                 lastSeenLeaderEpochs.put(tp, newEpoch);
                 return Optional.of(partitionMetadata);
+            } else if (changedTopicId) {
```

Comment:
I thought we changed the order of this in the 3.0 patch. We should be checking for a changed topic id before comparing epochs.